app/SmartAlbums/BaseSmartAlbum.php

```diff
@@ -39,7 +40,7 @@
  * Photos belong to these albums due to certain properties like being
  * starred, being recently added, etc.
  */
-abstract class BaseSmartAlbum implements AbstractAlbum
+abstract class BaseSmartAlbum extends Model implements AbstractAlbum
```

Comment:
but isn't this kind of expected? (I mean `extends Model`) 

because the extra logic in  the constructor of (`BaseSmartAlbum`) is doing some database stuff and it kinda expect this class to be the Model no?

also the `app/Models/Extensions/BaseAlbum.php` already extends `Model`, 

the structure is  supposed to be 

```
             AbstractAlbum
                 |
|---------------|------------------|
BaseAlbum                       BaseSmartAlbum 
```

no? or am I doing something else?